### ISSUE TITLE
InvalidEngineError: Cannot connect to the WebSocket server (reason: undefined)

### ISSUE THREAD

**Issue by gotheflu** [CheshireCaat/puppeteer-with-fingerprints#129]
I am getting InvalidEngineError while `const fingerprint = await plugin.fetch({ tags: ['Desktop'] })` is running after waiting for these:

```
The browser is downloading - this may take some time.
The browser is installing - this may take some time.
```

This is the error after browser installation is done (or stopped?):

```
InvalidEngineError: Cannot connect to the WebSocket server (reason: undefined)
This could be due to the fact that the engine was not downloaded or unpacked correctly.
Try completely deleting the engine folder and restarting the code until it completes.
If this does not help, open an issue with a detailed description of the problem.
    at C:\Users\User\Projects\newproject\node_modules\bas-remote-node\src\services\socket.js:51:18
    at Timeout._onTimeout (C:\Users\User\Projects\newproject\node_modules\chnl\dist\channel.cjs.js:2:4848)
    at listOnTimeout (node:internal/timers:594:17)
    at process.processTimers (node:internal/timers:529:7)
```

I am running this with paid Bablosoft key, on Windows Server 2019. I have tried reinstalling my npm packages and deleting the `data` folder multiple times. This setup was working for me before but not anymore. There is more than enough space in my disk. Here are some version info:

```powershell
PS> node -v
v22.8.0
PS> npm -v
10.8.2
```

```json
  "dependencies": {
    "puppeteer": "^22.15.0",
    "puppeteer-with-fingerprints": "^1.6.3"
  },
```

**Comment by denemegrafik** [comment]
I have created a new project with this code:

```js
const { plugin } = require("puppeteer-with-fingerprints");

plugin.setServiceKey("");

(async () => {
  const fingerprint = await plugin.fetch({
    tags: ["Microsoft Windows", "Chrome"],
  });

  plugin.useFingerprint(fingerprint);

  const browser = await plugin.launch({ headless: false });

  const page = await browser.newPage();
  await page.goto("https://example.com");

  console.log(
    "Viewport:",
    await page.evaluate(() => ({
      deviceScaleFactor: window.devicePixelRatio,
      width: document.documentElement.clientWidth,
      height: document.documentElement.clientHeight,
    }))
  );

  await browser.close();
})();
```

And the issue persists:

```powershell
PS> node .\index.js
The browser is downloading - this may take some time.
The browser is installing - this may take some time.
C:\Users\User\Projects\puptest\node_modules\bas-remote-node\src\services\socket.js:51
          reject(new InvalidEngineError(`Cannot connect to the WebSocket server (reason: ${error.reason})`));
                 ^

InvalidEngineError: Cannot connect to the WebSocket server (reason: undefined)
This could be due to the fact that the engine was not downloaded or unpacked correctly.
Try completely deleting the engine folder and restarting the code until it completes.
If this does not help, open an issue with a detailed description of the problem.
    at C:\Users\User\Projects\puptest\node_modules\bas-remote-node\src\services\socket.js:51:18
    at Timeout._onTimeout (C:\Users\User\Projects\puptest\node_modules\chnl\dist\channel.cjs.js:2:4848)
    at listOnTimeout (node:internal/timers:594:17)
    at process.processTimers (node:internal/timers:529:7)

Node.js v22.8.0
```

**Comment by CheshireCaat** [comment]
Please try using an earlier version of the plugin (for example **1.5.0**), if you say that it worked before, then we can try to understand what could have influenced the occurrence of the error.

Among other things, there is a chance that an antivirus or firewall may block the engine's operation - please check this point too. The error is related to the websocket, it cannot connect to the engine (it takes 60 attempts and 1 minute). Maybe a firewall, network settings, security policy, or something else prevents you from using the ports generated for websocket or it does not allow the engine to start at all.

**Comment by gotheflu** [comment]
First tried version 1.5.0 and reverted recently added `plugin.setServiceKey` method but the issue persisted.

Windows defender firewall is disabled at the moment. Also disabled virus protection options from Windows Defender but unfortunately the issue persists. There is no other antivirus software running.

On a relatively unrelated note, one of the things I did between the last successfully working state of the project and now was uninstall Windows IIS (Internet Information Services) and install Nginx via Chocolatey. I didn’t think it would cause any port-related issues, but to be safe, I stopped the Nginx service. However, the issue still persists.

**Comment by CheshireCaat** [comment]
If you have the opportunity, you can send the server data to the mail likangt2012@gmail.com, I would try to look for the reason.

**Comment by gotheflu** [comment]
I could finally pinpoint the issue by debugging different configurations after formatting the server to the factory settings once or twice and skipping some the steps I took every time. It turns out the `openssl` package I installed through Chocolatey to generate self-signed certificates was causing this problem even if I don't use it. I uninstalled the program for now and it works now. I don't really know exactly why it would cause this issue, I would be happy to read through you guys brainstorming. Anyway, thanks @CheshireCaat for the effort.
